react/features/stream-effects/blur/JitsiStreamPresenterEffect.js

```diff
@@ -0,0 +1,126 @@
+import {
+    CLEAR_INTERVAL,
+    INTERVAL_TIMEOUT,
+    SET_INTERVAL,
+    timerWorkerScript
+} from './TimerWorker';
+
+export default class JitsiStreamPresenterEffect {
+
+    /**
+     * Represents a modified MediaStream that adds a camera track at the 
+     * bottom right corner of the desktop track using a HTML canvas.
+     * <tt>JitsiStreamCanvasEffect</tt> does the processing of the original
+     * video stream.
+     */
+    constructor(videoStream) {
+        const videoDiv = document.createElement('div');
+        this._canvas = document.createElement('canvas');
+        const firstVideoTrack = videoStream.getVideoTracks()[0];
+        const { height, width, frameRate } = firstVideoTrack.getSettings
+            ? firstVideoTrack.getSettings()
+            : firstVideoTrack.getConstraints();
+
+        // Workaround for FF issue https://bugzilla.mozilla.org/show_bug.cgi?id=1388974
+        this._ctx = this._canvas.getContext('2d');
+        document.body.appendChild(this._canvas);
+        
+        this._desktopElement = document.createElement('video');
+        this._videoElement = document.createElement('video');
+        document.body.appendChild(videoDiv);
+        videoDiv.appendChild(this._videoElement);
+        videoDiv.appendChild(this._desktopElement);
+
+        this._frameRate = frameRate;
+        this._videoHeight = height;
+        this._videoWidth = width;
+        this._videoElement.width = width;
+        this._videoElement.height = height;
+        this._videoElement.autoplay = true;
+        this._videoElement.srcObject = videoStream;
+
+        // set the style attribute of the div to make it invisible
+        videoDiv.setAttribute('style', 'display:none');
+
+        // Bind event handler so it is only bound once for every instance.
+        this._onVideoFrameTimer = this._onVideoFrameTimer.bind(this);
+        this._videoFrameTimerWorker = new Worker(timerWorkerScript);
+        this._videoFrameTimerWorker.onmessage = this._onVideoFrameTimer;
+    }
+
+    /**
+     * EventHandler onmessage for the videoFrameTimerWorker WebWorker.
+     *
+     * @private
+     * @param {EventHandler} response - The onmessage EventHandler parameter.
+     * @returns {void}
+     */
+    _onVideoFrameTimer(response) {
+        if (response.data.id === INTERVAL_TIMEOUT) {
+            this._renderVideo();
+        }
+    }
+
+    /**
+     * Loop function to render the video frame input and draw blur effect.
+     *
+     * @private
+     * @returns {void}
+     */
+    _renderVideo() {
+        // flip the image horizontally before drawing it on the desktop image
```

Comment:
don't leave commented code in, please

Reply:
I am working on this still, will have it in the final PR.